File: backend/src/main/java/com/corusoft/ticketmanager/tickets/controllers/dtos/CreateTicketParamsDTO.java
package com.corusoft.ticketmanager.tickets.controllers.dtos;

import com.corusoft.ticketmanager.common.JacksonLocalDateDeserializer;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import jakarta.validation.constraints.*;
import lombok.*;

import java.time.LocalDate;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CreateTicketParamsDTO {
    @NotNull
    private Long userID;

    @NotBlank
    private String supplier;

    @NotNull
    private Long categoryID;

    @PastOrPresent
    @JsonDeserialize(using = JacksonLocalDateDeserializer.class)
    private LocalDate emmitedAtDate;

    @NotBlank
    private String emmitedAtTime;

    @NotBlank
    private String country;

    @NotBlank
    private String currency;

    @PositiveOrZero
    private Float totalTax;

    @PositiveOrZero
    private Float totalAmount;

    /** Ticket codificado como String en Base64 */
    @NotBlank
    private String ticketData;

    @NotBlank
    private String name;
}

File: mobile/app/src/main/java/com/corusoft/ticketmanager/backend/dtos/tickets/CreateTicketParamsDTO.kt
package com.corusoft.ticketmanager.backend.dtos.tickets

import java.time.LocalDate


data class CreateTicketParamsDTO(
    val userID: Long,
    val supplier: String,
    val categoryID: Long,
    val emmitedAtDate: LocalDate?,
    val emmitedAtTime: String,
    val country: String,
    val currency: String,
    val totalTax: Float,
    val totalAmount: Float,
    val ticketData: String,
    val name: String
)
